zrml/neo-swaps/src/lib.rs

```diff
@@ -1005,6 +1005,26 @@ mod pallet {
                 external_fee_percentage.mul_floor(ZeitgeistBase::<BalanceOf<T>>::get()?);
             swap_fee.checked_add_res(&external_fee_fractional)
         }
+
+        fn match_failure(error: DispatchError) -> ApiError<AmmSoftFail> {
+            let spot_price_too_low: DispatchError =
+                Error::<T>::NumericalLimits(NumericalLimitsError::SpotPriceTooLow).into();
+            let spot_price_slipped_too_low: DispatchError =
+                Error::<T>::NumericalLimits(NumericalLimitsError::SpotPriceSlippedTooLow).into();
+            let max_amount_exceeded: DispatchError =
+                Error::<T>::NumericalLimits(NumericalLimitsError::MaxAmountExceeded).into();
+            let min_amount_not_met: DispatchError =
+                Error::<T>::NumericalLimits(NumericalLimitsError::MinAmountNotMet).into();
+            if spot_price_too_low == error
+                || spot_price_slipped_too_low == error
+                || max_amount_exceeded == error
+                || min_amount_not_met == error
+            {
+                ApiError::SoftFailure(AmmSoftFail::Numerical)
+            } else {
+                ApiError::HardFailure(error)
+            }
```

Comment:
I'd make this a match statement.

Reply:
I wanted this so bad too.. But I found no working way to do it. I would be pleased if you can make a suggestion that works :-) Thanks a lot!

Reply:
Ah yeah, nevermind. I thought this was possible, but apparently I don't understand how match statements work.